src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -519,6 +526,10 @@ export class KycService {
     await this.stepLogRepo.save(entity);
   }
 
+  private async getKycClients(userData: UserData): Promise<Wallet[]> {
+    return Promise.all(userData.kycClientList?.map((k) => this.walletService.getByIdOrName(k)) ?? []);
+  }
```

Comment:
It's probably faster to fetch all wallets and do the filtering (with userData.kycClientList) in the mapper.